pkg/controller/backup_sync_controller_test.go

```diff
@@ -154,3 +158,133 @@ func TestBackupSyncControllerRun(t *testing.T) {
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupC").WithPhase(v1.BackupPhaseCompleted),
+		// 	},
+		// numMacthingBackups: 2,
+		// },
+		// {
+		// 	name:      "all overlapping backups",
+		// 	namespace: "ns-1",
+		// 	cloudBackups: []*v1.Backup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").Backup,
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").Backup,
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").Backup,
+		// 	},
+		// 	k8sBackups: []*arktest.TestBackup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").WithPhase(v1.BackupPhaseCompleted),
+		// 	},
+		// 	numMacthingBackups: 3,
+		// },
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				bs              = &arktest.BackupService{}
+				client          = fake.NewSimpleClientset()
+				sharedInformers = informers.NewSharedInformerFactory(client, 0)
+				logger          = arktest.NewLogger()
+			)
+
+			c := NewBackupSyncController(
+				client.ArkV1(),
+				bs,
+				"bucket",
+				time.Duration(0),
+				test.namespace,
+				sharedInformers.Ark().V1().Backups(),
+				logger,
+			).(*backupSyncController)
+
+			expectedActions := make([]core.Action, 0)
+
+			// setup: insert backups into Kubernetes
+			for _, backup := range test.k8sBackups {
+				actionCreate := core.NewCreateAction(
+					v1.SchemeGroupVersion.WithResource("backups"),
+					test.namespace,
+					backup.Backup,
+				)
+				expectedActions = append(expectedActions, actionCreate)
+
+				actionDelete := core.NewDeleteAction(
+					v1.SchemeGroupVersion.WithResource("backups"),
+					test.namespace,
+					backup.Name,
+				)
+				expectedActions = append(expectedActions, actionDelete)
+
+				// add test backup to informer:
+				err := sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(backup.Backup)
+				assert.NoError(t, err, "Error adding backup to informer")
+
+				// add test backup to kubernetes:
+				_, err = client.Ark().Backups(test.namespace).Create(backup.Backup)
+				assert.NoError(t, err, "Error deleting from clientset")
+
+			}
+
+			// get names of client backups
+			testBackupNames := sets.NewString()
+			for _, cloudBackup := range test.cloudBackups {
+				testBackupNames.Insert(cloudBackup.Name)
+			}
+
+			c.deleteUnused(testBackupNames)
+
+			numBackups, err := numBackups(t, c) // this is returning 3, but all backups should've been deleted since the test case doesn't have overlapping backups
```

Comment:
The issue here is that, because the **informer** isn't actually *running* in this test, it's not being updated when the sync controller issues a Delete call to the **client**.  The easiest thing to do would be to change `numBackups` to use the **client** to list backups, e.g. `client.Ark().Backups(test.Namespace).List(...)`

Reply:
Or maybe not? Catching up on slack real quick, h/o before processing this comment :)

Reply:
OK I think both this and the comment below are at play here